craft_parts/plugins/ant_plugin.py

```diff
@@ -142,6 +144,16 @@ def get_build_packages(self) -> Set[str]:
     @override
     def get_build_environment(self) -> Dict[str, str]:
         """Return a dictionary with the environment to use in the build step."""
+        # Getting ant to use a proxy requires a little work; the JRE doesn't
+        # help as much as it should.  (java.net.useSystemProxies=true ought
+        # to do the trick, but it relies on desktop configuration rather
+        # than using the standard environment variables.)
+        ant_opts = []  # type: List[str]
+        ant_opts.extend(_get_proxy_options("http"))
+        ant_opts.extend(_get_proxy_options("https"))
+        if ant_opts:
+            opts = " ".join(opt.replace("'", "'\\''") for opt in ant_opts)
+            return {"ANT_OPTS": opts}
```

Comment:
The maven plugin deals with this by updating an XML file. Could we unify these by putting this implementation into the `JavaPlugin` base class and setting the `_JAVA_OPTIONS` environment variable instead?

Reply:
I would ask this to be done as a separate PR.
I have created this PR as the bug I opened in Snapcraft was not getting any attention and with 18.04 now being in ESM, we need to provide an upgrade path to any `base: core18` snaps to move to `base core22`. Currently, any ant builds fail in Launchpad because of missing proxy support.

Reply:
So it seems that this also needs to be added as a hotfix to 1.26.